group/serializers.py

```diff
@@ -1,11 +1,31 @@
 """Serializer module"""
 from rest_framework import serializers
+from django.contrib.auth.models import User
+from user.serializers import UserSerializer
 from .models import Group
 
 
 class GroupSerializer(serializers.ModelSerializer):
-    """Group seerializer class"""
+    """Group serializer class"""
+
     class Meta:
         """Serializer Meta class"""
         model = Group
         fields = ['id', 'name', 'members', 'creator']
+
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        representation['members'] = UserSerializer(
+                                    instance.members, many=True).data
+        return representation
+
+    def update(self, instance, validated_data):
+        existing_members = instance.members.all()
+        new_members = validated_data.get("members")
+
+        for member in new_members:
+            if member not in existing_members:
+                user = User.objects.get(id=member)
+                instance.members.add(user)
```

Comment:
From [this](https://docs.djangoproject.com/en/5.0/topics/db/examples/many_to_many/) documentation, this is redundant. Django under the hood takes care of managing duplicates in m2m relationships. We do not need to check if it exists before adding. Propose to get all member objects from the ids provided, and then do a batch or bulk add at once.